Add etymology-lookup

clairefro opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/clairefro/obsidian-plugin-etymology-lookup

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [x]  Linux
  <Tested mobile with emulator only>
  - [ ]  Android _(if applicable)_ 
  - [ ]  iOS _(if applicable)_ 
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
 - [x] I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[console.log(selected);](https://github.com/clairefro/obsidian-plugin-etymology-lookup/blob/3dc7330fa602362f218ccac464ebcc0dc02b272d/main.ts#L75)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[const activeLeaf = this.app.workspace.activeLeaf;](https://github.com/clairefro/obsidian-plugin-etymology-lookup/blob/3dc7330fa602362f218ccac464ebcc0dc02b272d/main.ts#L91)
It's highly discouraged that you use the global app instance in your plugin. It's currently accessible from the global context for debugging purposes, and might be removed in the future.
Instead, you should use the reference to App that you have from your plugin instance.
`this` in this case is referring to the global scope.
Also, `activeLeaf` is deprecated, to get the currently active editor use `this.app.workspace.activeEditor?.editor;`

clairefro commented:
@joethei Thank you 

I fixed these and released new version `1.0.1`
- removed console.log()
- replaced deprecated API `activeLeaf` -> `activeEditor`

And also added ability to run command with right-click menu option

joethei commented:
You missed the first part of the second issue:

[const editor = this.app.workspace.activeEditor?.editor;](https://github.com/clairefro/obsidian-plugin-etymology-lookup/blob/bbad4a570dd764372870f7f8a4ac4eb11ffa8a6e/main.ts#L93C2-L93C57)
It's highly discouraged that you use the global app instance in your plugin. It's currently accessible from the global context for debugging purposes, and might be removed in the future.
Instead, you should use the reference to App that you have from your plugin instance.
this in this case is referring to the global scope.

clairefro commented:
@joethei thank you 

Moved app context from global to plugin in [this commit](https://github.com/clairefro/obsidian-plugin-etymology-lookup/commit/7565e2acf00788332bd7c923d9a368041d931ce5) for 1.0.2 release
